### PR TITLE
Add  fixtureList viewmodel and fix blank GW bug

### PR DESCRIPTION
**Pull request by HoinzeyBear**
Fix bug where after loading and going to a blank gameweek (GW7) we were showing placeholder data rather than an empty list

### CODE REVIEW

File: app/src/main/java/dev/johnoreilly/fantasypremierleague/di/AppModule.kt [hunk 1]
```diff
@@ -1,9 +1,11 @@
 package dev.johnoreilly.fantasypremierleague.di
 
 import dev.johnoreilly.fantasypremierleague.presentation.FantasyPremierLeagueViewModel
+import dev.johnoreilly.fantasypremierleague.presentation.fixtures.FixturesViewModel
 import org.koin.androidx.viewmodel.dsl.viewModel
 import org.koin.dsl.module
 
 val appModule = module {
     viewModel { FantasyPremierLeagueViewModel(get()) }
+    viewModel { FixturesViewModel() }
```
Comment (HoinzeyBear) [inline app/src/main/java/dev/johnoreilly/fantasypremierleague/di/AppModule.kt:10]: Complete shot in the dark, haven't spent any time working out Realm or Koin

File: app/src/main/java/dev/johnoreilly/fantasypremierleague/presentation/fixtures/FixturesViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package dev.johnoreilly.fantasypremierleague.presentation.fixtures
+
+import androidx.lifecycle.ViewModel
+import dev.johnoreilly.common.data.repository.FantasyPremierLeagueRepository
+import kotlinx.coroutines.flow.StateFlow
+import org.koin.java.KoinJavaComponent.inject
+
+class FixturesViewModel: ViewModel() {
+
+    private val repository: FantasyPremierLeagueRepository by inject(FantasyPremierLeagueRepository::class.java)
```
Comment (joreilly) [inline app/src/main/java/dev/johnoreilly/fantasypremierleague/presentation/fixtures/FixturesViewModel.kt:10]: you can pass `repository` in the constructor (as `FantasyPremierLeagueViewModel` is (the Koin `get()` will automatically inject it)

File: app/src/main/java/dev/johnoreilly/fantasypremierleague/presentation/fixtures/FixturesListView.kt [hunk 3]
```diff
@@ -16,19 +16,20 @@ import androidx.compose.ui.Modifier
...
 import dev.johnoreilly.fantasypremierleague.presentation.global.maroon200
 
 @Composable
 fun FixturesListView(
-    fantasyPremierLeagueViewModel: FantasyPremierLeagueViewModel,
     onFixtureSelected: (fixtureId: Int) -> Unit,
 ) {
-    val fixturesState = fantasyPremierLeagueViewModel.gameweekToFixtures.collectAsState()
-    val currentGameweek: State<Int> = fantasyPremierLeagueViewModel.currentGameweek.collectAsState()
+    val fixturesViewModel: FixturesViewModel = viewModel()
```
Comment (joreilly) [inline app/src/main/java/dev/johnoreilly/fantasypremierleague/presentation/fixtures/FixturesListView.kt:29]: A pattern I'm using in other projects (and should have done so here as well) is pass view model in as parameter (with default value of `getViewMode()` as is done in following for example https://github.com/joreilly/PeopleInSpace/blob/main/app/src/main/java/com/surrus/peopleinspace/personlist/PersonListScreen.kt#L60 ).

Comment (joreilly) [reply app/src/main/java/dev/johnoreilly/fantasypremierleague/presentation/fixtures/FixturesListView.kt:29]: This seems to be approach generally used in various compose samples as it makes it easier to test the composable.

Comment (joreilly) [reply app/src/main/java/dev/johnoreilly/fantasypremierleague/presentation/fixtures/FixturesListView.kt:29]: Yeah, I think that's preferable all right.  Often maybe be a top level screen composable might know about the viewmodel (or at least it's in constructor like that) and then pass only data down to lower level composables.  I like approach in PeopleInSpace (stolen from NowInAndroid!) where "route" composable does this.

Comment (HoinzeyBear) [reply app/src/main/java/dev/johnoreilly/fantasypremierleague/presentation/fixtures/FixturesListView.kt:29]: Yes that is exactly how it is done in Joe Birch's course too.

Comment (HoinzeyBear) [reply app/src/main/java/dev/johnoreilly/fantasypremierleague/presentation/fixtures/FixturesListView.kt:29]: Why doesn't that PeopleInSpace example throw an exception for not having a zero-argument-constructor for the ViewModel ?

Comment (joreilly) [reply app/src/main/java/dev/johnoreilly/fantasypremierleague/presentation/fixtures/FixturesListView.kt:29]: Which code is this?

Comment (HoinzeyBear) [reply app/src/main/java/dev/johnoreilly/fantasypremierleague/presentation/fixtures/FixturesListView.kt:29]: >  pass view model in as parameter (with default value of getViewMode() as is done in following for example https://github.com/joreilly/PeopleInSpace/blob/main/app/src/main/java/com/surrus/peopleinspace/personlist/PersonListScreen.kt#L60 )

[That VM takes in a repo](https://github.com/joreilly/PeopleInSpace/blob/8463a3c84e81c1b0988f2ca9d3363f92b4634119/app/src/main/java/com/surrus/peopleinspace/personlist/PersonListViewModel.kt#L20)

If I copy that though, I get an error for my VM having no zero-arg constructor. I might pull down PIS and compare

Comment (joreilly) [reply app/src/main/java/dev/johnoreilly/fantasypremierleague/presentation/fixtures/FixturesListView.kt:29]: that's injected by Koin (based on what's in `AppModule`

**Comment by joreilly** [comment]
closing this to create new PR with conflicts fixed and some additional updates https://github.com/joreilly/FantasyPremierLeague/pull/88

**Pull request closed without merging**